// @redocly/cli/lib/__mocks__/@redocly/openapi-core.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.doesYamlFileExist = exports.findConfig = exports.slash = exports.formatProblems = exports.getTotals = exports.bundle = exports.lint = exports.getMergedConfig = exports.loadConfig = exports.RedoclyClient = exports.__redoclyClient = void 0;
const config_1 = require("./../../__tests__/fixtures/config");
exports.__redoclyClient = {
    isAuthorizedWithRedocly: jest.fn().mockResolvedValue(true),
    isAuthorizedWithRedoclyByRegion: jest.fn().mockResolvedValue(true),
    login: jest.fn(),
    registryApi: {
        setAccessTokens: jest.fn(),
        authStatus: jest.fn(),
        prepareFileUpload: jest.fn().mockResolvedValue({
            signedUploadUrl: 'signedUploadUrl',
            filePath: 'filePath',
        }),
        pushApi: jest.fn(),
    },
};
exports.RedoclyClient = jest.fn(() => exports.__redoclyClient);
exports.loadConfig = jest.fn(() => config_1.ConfigFixture);
exports.getMergedConfig = jest.fn();
exports.lint = jest.fn();
exports.bundle = jest.fn(() => ({ bundle: { parsed: null }, problems: null }));
exports.getTotals = jest.fn(() => ({ errors: 0 }));
exports.formatProblems = jest.fn();
exports.slash = jest.fn();
exports.findConfig = jest.fn();
exports.doesYamlFileExist = jest.fn();
